solutions/USACO jan20-wormholesort.mdx

```diff
@@ -2,15 +2,203 @@
 id: usaco-992
 source: USACO Silver 2020 January
 title: Wormhole Sort
-author: Óscar Garries
+author: Óscar Garries, Neo Wang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_wormsort_silver_jan20.html)
 
+## Alternative Solution (DSU/UF + Binary Search)
+
+Like the DFS solution, we binary search on the answer $x$. $x$ is valid if all $p_i$ are in the same component as $i$, which we can query in $\mathcal{O}(\alpha(N))$ using a Union Find/Disjoint Set data structure.
+
+Our resulting time complexity is $\mathcal{O}(N\log N)$
+
+## Alternative Solution (DSU/UF)
+
+We can alternatively search for our weight by adding the weights greatest $\rightarrow$ least until all $p_i$ are in the same component as $i$.
+
+Our resulting time complexity is $\mathcal{O}(N + M\log M)$ because of our initial sorting of the weights.
+
 <LanguageSection>
 <CPPSection>
 
-## C++ Implementation
+## C++ Implementation (DSU/UF + Binary Search)
```

Comment:
Implementations should be third-tier headings (###) under the two-tier headings (##).